app/src/main/java/vn/sunasterisk/moviedb_40/data/source/MovieDataSource.kt

```diff
@@ -0,0 +1,40 @@
+package vn.sunasterisk.moviedb_40.data.source
+
+import androidx.databinding.ObservableArrayList
+import io.reactivex.Observable
+import vn.sunasterisk.moviedb_40.data.model.Actor
+import vn.sunasterisk.moviedb_40.data.model.Movie
+import vn.sunasterisk.moviedb_40.data.source.remote.response.GenreResponse
+import vn.sunasterisk.moviedb_40.data.source.remote.response.MovieResponse
+
+interface MovieDataSource {
+    interface Local {
+        val allFavorite: ObservableArrayList<Movie>
+
+        fun addFavorite(movie: Movie): Boolean
+
+        fun deleteFavorite(movie: Movie): Boolean
+
+        fun isFavorite(movieId: Int): Boolean
+    }
+
+    interface Remote {
+        fun getGenres(): Observable<GenreResponse>
+
+        fun getMoviesTrending(): Observable<MovieResponse>
```

Comment:
```suggestion
        fun getMoviesTrending(): Observable<ListMovieResponse>
```
E nên có một model là ListMovieReponse chưa một list MovieReponse